### ISSUE TITLE
Transition all core pipelines to `1es-templates`

### ISSUE THREAD

**Issue by scbedd** [Azure/azure-sdk-tools#7717]
## Pre-req work

- Prototype a working conversion of the primary `stages` templates in each repo. Ensure `release` works on each of these. Ensure that the builds are green without failing injected steps.
   - [x] `azure/azure-sdk-for-python`
      - [Changeset](https://github.com/Azure/azure-sdk-for-python/compare/1es-template-conversion?expand=1) 
   - [x] `azure/azure-sdk-for-net`
      - [Changeset](https://github.com/Azure/azure-sdk-for-net/compare/main...benbp/pipeline-template-automatic) 
   - [x] `azure/azure-sdk-for-java`
      - [Changeset](https://github.com/Azure/azure-sdk-for-java/compare/1es-template-conversion?expand=1) 
   - [x] `azure/azure-sdk-for-js`
      - [Changeset](https://github.com/Azure/azure-sdk-for-js/compare/1es-template-conversion?expand=1) 
   - [x] `azure/azure-sdk-for-go`
      - [Changeset](https://github.com/Azure/azure-sdk-for-go/compare/1es-template-conversion?expand=1)
   - [x] `azure/azure-sdk-for-c`
     - [ChangeSet](https://github.com/Azure/azure-sdk-for-c/compare/1es-template-conversion?expand=1)
   - [x] `azure/azure-sdk-for-cpp`
     - [ChangeSet](https://github.com/Azure/azure-sdk-for-cpp/compare/1es-template-conversion?expand=1)
   - [x] `azure/azure-sdk-for-ios`
     - [ChangeSet](https://github.com/Azure/azure-sdk-for-ios/compare/1es-template-conversion?expand=1)
   - [ ] `azure/azure-sdk-for-android`
   - [ ] `azure/azure-sdk`

## Rollout 

There are necessary `eng/common` changes that need to roll out with these changes. To ensure easy rollout without breaking compatibility, perhaps the necessary pool and job changes should be done _alongside_ the `eng/common` changes. That means

1. ~~Submit `eng/common` changes to `azure/azure-sdk-tools`. ~~
2. ~~Let `eng/common` pipelines submit PRs to each repo~~
3. ~~Apply changes from `1es-template-conversion` branches from each repo _to_ the `eng/common` branch.~~
4. ~~Ensure all pipelines pass the common `template` release from each `eng/common` PR branch.~~
5. ~~Merge them all, following the eng/common merge strategy for `azure-sdk-tools` as well.~~

The above was judged too risky. Decided to go with a bit slower, but safer approach of:

1. Submit PR to eng/common w/ new generation logic (hardcoded OS) and 1es publish
2. Work through repos one at a time, integrating the new eng/common changes into `1es-template-conversion`

| Repo | code complete | template release | template livetests | `core` run | merged |
|---|---|---|---|---|---|
| `azure/azure-sdk-for-python`  |✅|[✅](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3557169)|[✅](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3556957)|[✅](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3560998&view=results)|✅|
| `azure/azure-sdk-for-net`     |✅|✅|✅|✅|✅|
| `azure/azure-sdk-for-java`    |✅|✅|✅|✅|✅|
| `azure/azure-sdk-for-js`      |✅|[✅](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3571404&view=results)|[✅](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3571436&view=results)|[✅](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3577021&view=results)|✅|
| `azure/azure-sdk-for-go`      |✅|[✅](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3594407&view=results)|✅|[✅](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3594371&view=results)|✅|
| `azure/azure-sdk-for-c`       |✅|✅|✅|✅|✅|
| `azure/azure-sdk-for-cpp`     |✅|[✅](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3585132&view=logs&j=8eaffc5a-ef20-56fd-55bd-1759aa97a456)|✅|[✅](https://dev.azure.com/azure-sdk/internal/_build?definitionId=1615)|✅|
| `azure/azure-sdk-for-ios`     |✅|✅|✅|✅|✅|

> **Note**: `azure-sdk-for-net` has merge blocked on a 1es-templates limitation.

## `Designer` Builds

The following builds are `designer`, and will not benefit from the `yml` updates above.

- `mgmt-netsdk-sign`
- `mgmt-NetCore-SDK-Publish`
- `mgmt-netsdk-multiapi-publish`
- `mgmt-netsdk-sdkcommon-publish`
- `mgmt-netsdk-publish`
- `Specs Repo - Update readme.md all-api-versions tags.`
- `Autorest Npm Admin Task`
- `tools - Codex - vscode.dev`

We need to come up with a strategy for these.

## `Release` Builds

Need to come up with a strategy for all the `internal` release pipelines that do not have yaml attached to them. [Full List](https://dev.azure.com/azure-sdk/internal/_release)

**Comment by scbedd** [comment]
Creating a new issue for long tail items.
